src/Android/Xamarin.Android/Samples/Augmented Reality/CollectDataAR/CollectDataAR.cs

```diff
@@ -0,0 +1,530 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Android;
+using Android.App;
+using Android.Content;
+using Android.Content.PM;
+using Android.Media;
+using Android.OS;
+using Android.Runtime;
+using Android.Support.V4.App;
+using Android.Support.V4.Content;
+using Android.Views;
+using Android.Widget;
+using ArcGISRuntimeXamarin.Samples.ARToolkit.Controls;
+using Esri.ArcGISRuntime.ARToolkit;
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.Tasks.NetworkAnalysis;
+using Esri.ArcGISRuntime.UI;
+using Esri.ArcGISRuntime.UI.Controls;
+using Java.Nio;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Surface = Esri.ArcGISRuntime.Mapping.Surface;
+
+namespace ArcGISRuntimeXamarin.Samples.CollectDataAR
+{
+    [Activity(ConfigurationChanges = Android.Content.PM.ConfigChanges.Orientation | Android.Content.PM.ConfigChanges.ScreenSize)]
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Collect data in AR",
+        "Augmented reality",
+        "Use a route displayed in the real world to navigate.",
+        "")]
+    public class CollectDataAR : Activity, IDialogInterfaceOnCancelListener
+    {
+        // Hold references to UI controls.
+        private ARSceneView _arView;
+        private TextView _helpLabel;
+        private Button _calibrateButton;
+        private Button _addButton;
+        private Button _roamingButton;
+        private Button _localButton;
+        private View _calibrationView;
+        private JoystickSeekBar _headingSlider;
+        private JoystickSeekBar _altitudeSlider;
+
+        // Static field for sharing route between views.
+        public static RouteResult PassedRouteResult;
+
+        // Scene content.
+        private ArcGISTiledElevationSource _elevationSource;
+        private Surface _elevationSurface;
+        private Scene _scene;
+
+        // Track when user is changing between AR and GPS localization.
+        private bool _changingScale;
+
+        // Create a new copmletion source for the prompt.
+        private TaskCompletionSource<int> _healthCompletionSource;
+
+        // Feature table for collected data about trees.
+        private ServiceFeatureTable _featureTable = new ServiceFeatureTable(new Uri("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/AR_Tree_Survey/FeatureServer/0"));
```

Comment:
The feature layer should be shown so that it is clear to the user which data points have already been collected.